app/_src/kubernetes-ingress-controller/troubleshooting.md

```diff
@@ -310,58 +310,134 @@ the Helm chart), wait for the Deployment to restart, run `kubectl
-object that was associated with the failure. An event's message is populated with an
-explanation of the exact reason for the failure, which should help you identify the root cause.
+{{site.kic_product_name}}'s [Prometheus metrics](/kubernetes-ingress-controller/{{page.kong_version}}/references/prometheus)
+include `ingress_controller_translation_count` and
+`ingress_controller_configuration_push_count` counters. Issues that require
+human intervention add `success=false` tallies to these counters.
 
-### Example
-In the following example, we create a service with a single port number `80`. In the Ingress definition, we specify a backend
-service that refers to a port number `8080` which does not match the one defined in the service.
-{{site.kic_product_name}} will skip the affected path and record warning events for both the service and Ingress objects.
+{{site.kic_product_name}} will also emit error logs with a `could not update
+kong admin` for configuration push failures.
+
+### Finding problem resource Events
+
+Once you have observed a translation or configuration push failure, you can
+locate which Kubernetes resources require changes by searching for Events. For
+example, this Ingress attempts to create a gRPC route that also uses HTTP
+methods, which is impossible:
 
 ```yaml
-apiVersion: v1
-kind: Service
-metadata:
-  name: my-service
-spec:
-  ports:
-    - port: 80
----
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
-  name: my-ingress
   annotations:
-    kubernetes.io/ingress.class: "kong"
+    konghq.com/methods: GET
+    konghq.com/protocols: grpcs
+    kubernetes.io/ingress.class: kong
+  name: httpbin
+  uid: 3dcd75e9-c076-46a0-8cd3-3cc62f081920
 spec:
   rules:
-    - http:
-        paths:
-          - path: /
-            pathType: Prefix
-            backend:
-              service:
-                name: my-service
-                port:
-                  number: 8080 # doesn't match the port in my-service
+  - http:
+      paths:
+      - backend:
+          service:
+            name: httpbin
+            port:
+              number: 80
+        path: /bar
+        pathType: Prefix
+```
+
+{{site.base_gateway}} will reject the route {{site.kic_product_name}} creates
+from this Ingress and return an error. {{site.kic_product_name}} will process
+this error and create a Kubernetes Event linked to the Ingress.
+
+You can quickly find these Events by searching across all namespaces for Events
+with the special failure reasons that indicate {{site.kic_product_name}}
+failures:
+
+```bash```
+kubectl get events -A --field-selector='reason=KongConfigurationApplyFailed'
+```
+Response:
+```
+NAMESPACE   LAST SEEN   TYPE      REASON                         OBJECT            MESSAGE
+default     35m         Warning   KongConfigurationApplyFailed   ingress/httpbin   invalid methods: cannot set 'methods' when 'protocols' is 'grpc' or 'grpcs'
+```
```

Comment:
Since `KongConfigurationApplyFailed` only applies to 2.9.x or later, I'd add {% if_version gte:2.9.x %}  around this. Should we add an example that would display for only 2.8.x (`{% if_version eq:2.8.x %}`) for `KongConfigurationTranslationFailed`?